File: src/main/java/com/pubnub/api/models/consumer/objects_api/member/PNMember.java
package com.pubnub.api.models.consumer.objects_api.member;

import com.pubnub.api.models.consumer.objects_api.PNObject;
import com.pubnub.api.models.consumer.objects_api.user.PNUser;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@Getter
@EqualsAndHashCode(callSuper = true)
public class PNMember extends PNObject {

    private PNUser user;

    private PNMember() {
    }
}

File: src/main/kotlin/com/pubnub/api/models/consumer/objects/member/PNMember.kt
package com.pubnub.api.models.consumer.objects.member

import com.pubnub.api.models.consumer.objects.uuid.PNUUIDMetadata

data class PNMember(
    val uuid: PNUUIDMetadata?,
    val custom: Any? = null,
    val updated: String,
    val eTag: String,
    val status: String?
) {
    data class Partial(
        val uuidId: String,
        override val custom: Any? = null,
        override val status: String? = null,

    ) : MemberInput {
        override val uuid: String = uuidId
    }
}
